common/src/main/java/com/wynntils/core/chat/tabs/ChatTab.java

```diff
@@ -65,4 +66,21 @@ public Pattern getCustomRegex() {
                 ? customRegex = Pattern.compile(customRegexString, Pattern.DOTALL)
                 : customRegex;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) return true;
+        if (other == null || getClass() != other.getClass()) return false;
+
+        ChatTab chatTab = (ChatTab) other;
+        return consuming == chatTab.consuming
+                && Objects.equals(name, chatTab.name)
+                && Objects.equals(filteredTypes, chatTab.filteredTypes)
+                && Objects.equals(customRegexString, chatTab.customRegexString);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, consuming, filteredTypes, customRegexString);
+    }
```

Comment:
... ChatTabs really should have been a record. Too bad it does not play nicely with gson. :-(